From 3dfd4d93fa013e1c0578d3ceac5c8f4ebba4b6ec Mon Sep 17 00:00:00 2001
From: Matthias Braun <matze@braunis.de>
Date: Mon, 12 Sep 2022 09:17:38 -0700
Subject: [PATCH 1013/1021] ControlHeightReduction: Remove assert check in
 shouldApply

Remove assertion checking for non-empty `ProfileSummaryInfo`.

Differential Revision: https://reviews.llvm.org/D133706

(cherry picked from commit 6d972ad2d8f1cfe22a61d1c4aed6089f0586b377)
---
 llvm/lib/Transforms/Instrumentation/ControlHeightReduction.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/llvm/lib/Transforms/Instrumentation/ControlHeightReduction.cpp b/llvm/lib/Transforms/Instrumentation/ControlHeightReduction.cpp
index adc007dacae4..0d2662108e5b 100644
--- a/llvm/lib/Transforms/Instrumentation/ControlHeightReduction.cpp
+++ b/llvm/lib/Transforms/Instrumentation/ControlHeightReduction.cpp
@@ -406,7 +406,6 @@ static bool shouldApply(Function &F, ProfileSummaryInfo& PSI) {
     return CHRFunctions.count(F.getName());
   }
 
-  assert(PSI.hasProfileSummary() && "Empty PSI?");
   return PSI.isFunctionEntryHot(&F);
 }
 
-- 
2.39.0

